src/main/java/frc/robot/RobotContainer.java

```diff
@@ -52,34 +55,105 @@
   private JoystickButton[] mButtons = new JoystickButton[11];
 
   // subsystems
-  private static final DriveBaseSubsystem driveBaseSubsystem = new DriveBaseSubsystem(driverJoystick);
-  private final CDSSubsystem CDSSubsystem = new CDSSubsystem();
-  private final IntakeSubsystem IntakeSubsystem = new IntakeSubsystem(); 
-  private final ShooterSubsystem ShooterSubsystem = new ShooterSubsystem();
-  private final LimelightSubsystem LimelightSubsystem = new LimelightSubsystem();
+  private static DriveBaseSubsystem driveBaseSubsystem;
+  private static CDSSubsystem CDSSubsystem;
+  private static IntakeSubsystem IntakeSubsystem; 
+  private static ShooterSubsystem ShooterSubsystem;
+  private static LimelightSubsystem LimelightSubsystem;
 
   // commands
-  private final DriveBaseTeleopCommand mDriveBaseTeleopCommand = new DriveBaseTeleopCommand(driveBaseSubsystem);
-  private IntakeForwardCommand intakeForwardCommand = new IntakeForwardCommand(IntakeSubsystem);
-  private IntakeReverseCommand intakeReverseCommand = new IntakeReverseCommand(IntakeSubsystem);
-  private ShooterPrime shooterPrime = new ShooterPrime(ShooterSubsystem,LimelightSubsystem);
-  private CDSForwardCommand CDSForwardCommand = new CDSForwardCommand(CDSSubsystem);
-  private CDSReverseCommand CDSReverseCommand = new CDSReverseCommand(CDSSubsystem);
-  private LimelightAlign limelightAlign = new LimelightAlign(LimelightSubsystem,driveBaseSubsystem);
+  private DriveBaseTeleopCommand mDriveBaseTeleopCommand;
+  private IntakeForwardCommand intakeForwardCommand;
+  private IntakeReverseCommand intakeReverseCommand;
+  private ShooterPrime shooterPrime;
+  private CDSForwardCommand CDSForwardCommand;
+  private CDSReverseCommand CDSReverseCommand;
+  private LimelightAlign limelightAlign;
+
 
   // auton
   // private Trajectory[] mTrajectories;  // multiple trajectories
   // private int trajectoryIndex = 0;
   private Trajectory trajectory;
 
+  private JSONObject subSysEnables;
+
   // The container for the robot. Contains subsystems, OI devices, and commands.
   public RobotContainer() {
+    try {
+      String jsonString = Files.readString(Filesystem.getDeployDirectory().toPath().resolve("SubsystemEnables.json"));
+      JSONArray jsonArray = (JSONArray) JSONValue.parse(jsonString);
+      subSysEnables = (JSONObject) jsonArray.get(0);
+       
+      //for (Object k : subSysEnables.keySet()) {
+      //   System.out.println((String) k + " " + subSysEnables.get((String) k));
+      //}
+
+
+    } catch (IOException e) {
+      //TODO catch/fail gracefully
+    }
+    
+
+
     debugTab = Shuffleboard.getTab("debug");
+
     // Configure the button bindings
     for (int i = 1; i < mButtons.length; i++) {
       mButtons[i] = new JoystickButton(driverJoystick, i);
     }
 
+    // subsystems
+    for (Object k : subSysEnables.keySet()) {
+      if (((String)subSysEnables.get((String) k)).equals("true")) {
```

Comment:
"true".equals((String)subSysEnables.get((String) k))

if you change the iteration at all, then using the constant prevents NPE's